gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveDataset.java

```diff
@@ -174,39 +190,156 @@ private Path resolvePath(String pattern) {
 
     List<CopyableFile> copyableFiles = Lists.newArrayList();
 
-    try(AutoReturnableObject<IMetaStoreClient> client = new AutoReturnableObject<>(this.clientPool)) {
+    HiveMetastoreClientPool targetClientPool = HiveMetastoreClientPool.get(this.properties,
+        Optional.fromNullable(properties.getProperty(TARGET_METASTORE_URI_KEY)));
+    String targetDatabase = Optional.fromNullable(this.properties.getProperty(TARGET_DATABASE_KEY)).or(this.table.getDbName());
 
-      if (isPartitioned()) {
+    try(AutoReturnableObject<IMetaStoreClient> sourceClient = this.clientPool.getClient();
+        AutoReturnableObject<IMetaStoreClient> targetClient = targetClientPool.getClient()) {
 
-        List<Partition> partitions = Lists.newArrayList();
-        for (org.apache.hadoop.hive.metastore.api.Partition p : client.get().listPartitions(this.table.getDbName(),
-            this.table.getTableName(), (short) -1)) {
-          partitions.add(new Partition(this.table, p));
+      Optional<Table> targetTable = Optional.absent();
+      if (targetClient.get().tableExists(targetDatabase, this.table.getTableName())) {
+        targetTable = Optional.of(new Table(targetClient.get().getTable(targetDatabase, this.table.getTableName())));
+      }
+
+      if (targetTable.isPresent()) {
+        try {
+          checkTableCompatibility(this.table, targetTable.get());
+        } catch (IOException ioe) {
+          log.error("Source and target table are not compatible. Will not copy table.");
```

Comment:
I think we should have option, such as `force.copy`, to alter the destination table.